packages/experience-builder-sdk/src/core/preview/designComponentUtils.ts

```diff
@@ -1,7 +1,54 @@
-import { CompositionNode } from '../../types';
+import { CompositionComponentPropValue, CompositionNode } from '../../types';
 import { checkIfDesignComponent } from '../../utils/utils';
 import { EntityStore } from './EntityStore';
 
+export const deserializeDesignComponentNode = ({
+  node,
+  componentInstanceVariables,
+}: {
+  node: CompositionNode;
+  componentInstanceVariables: CompositionNode['variables'];
+}): CompositionNode => {
+  const variables: Record<string, CompositionComponentPropValue> = {};
+
+  for (const [variableName, variable] of Object.entries(node.variables)) {
+    variables[variableName] = variable;
+    if (variable.type === 'ComponentValue') {
+      const componentValueKey = variable.key;
+      const instanceProperty = componentInstanceVariables[componentValueKey];
+
+      // For design component, we look up the variable in the design component instance and
+      // replace the componentValue with that one.
+      if (instanceProperty?.type === 'UnboundValue') {
+        variables[variableName] = {
+          type: 'UnboundValue',
+          key: instanceProperty.key,
+        };
+      } else if (instanceProperty?.type === 'BoundValue') {
+        variables[variableName] = {
+          type: 'BoundValue',
+          path: instanceProperty.path,
+        };
+      }
+    }
+  }
+
+  const children: CompositionNode[] = node.children.map((child) =>
+    deserializeDesignComponentNode({
+      node: child,
+      componentInstanceVariables,
+    })
+  );
+
+  return {
+    // separate node type identifiers for design components and their blocks, so we can treat them differently in as much as we want
+    // type: isDesignComponent ? DESIGN_COMPONENT_NODE_TYPE : DESIGN_COMPONENT_BLOCK_NODE_TYPE,
```

Comment:
Do we still need to set the type here? I tested it and seems to work fine without it. Should this comment be removed?

Reply:
Yes, you're right. Thanks for pointing out :)